pyrealm/core/solar.py

```diff
@@ -1,26 +1,600 @@
+
+    .. math::
+
+        hs = \frac{\arccos(-1.0 \cdot \text{clip}(\frac{ru}{rv}, -1.0, 1.0))}{k\_pir}
+
+    Args:
+        delta       : solar declination delta
+        latitude    : site latitude(s)
+        k_pir       : constant rad to degrees conversion, degrees/rad
+
+    Returns:
+        An array of local hour angle, degrees
+    """
+
+    ru, rv = calc_lat_delta_intermediates(delta=delta, latitude=latitude)
+
+    return _calc_sunset_hour_angle_from_ru_rv(ru, rv, k_pir)
+
+
+def _calc_sunset_hour_angle_from_ru_rv(
+    ru: NDArray, rv: NDArray, k_pir: float
+) -> NDArray:
+    """Calculate sunset hour angle from intermediates.
+
+    This function calculates the sunset hour angle using Eq3.22,
+    :cite:t:'stine_geyer:2001'.
+
+    Args:
+        ru          : dimensionless parameter
+        rv          : dimensionless parameter
+        k_pir       : constant rad to degrees conversion, degrees/rad
+    Returns:
+        An array of local hour angle, degrees
+    """
+
+    return np.arccos(-1.0 * np.clip(ru / rv, -1.0, 1.0)) / k_pir
+
+
+def calc_daily_solar_radiation(
+    dr: NDArray, hs: NDArray, delta: NDArray, latitude: NDArray, const: CoreConst
+) -> NDArray:
+    r"""Calculate daily extraterrestrial solar radiation (J/m^2).
+
+    This function calculates the daily extraterrestrial solar radition (J/m^2)
+    using Eq. 1.10.3 :cite:t:'Duffie & Beckman:2013'.
+
+    .. math::
+
+        ra\_d = \left( \frac{secs\_d}{\pi} \right) \cdot rad\_const \cdot dr \cdot
+        \left(ru \cdot k\_pir \cdot hs + rv \cdot \sin(\deg2rad(hs))\right)
+
+    Args:
+        rad_const   : planetary radiation constant, W/m^2
+        dr          : dimensionless distance factor
+        hs          : local hour angle, degrees
+        delta       : solar declination delta
+        latitude    : site latitude(s)
+        const       : CoreConst object containing core constants:
+
+
+    Returns:
+        NDArray     : An array of daily solar radiation, J/m^2
+    """
+
+    ru, rv = calc_lat_delta_intermediates(delta=delta, latitude=latitude)
+
+    return _calc_daily_solar_radiation(dr=dr, ru=ru, rv=rv, hs=hs, const=const)
+
+
+def _calc_daily_solar_radiation(
+    dr: NDArray, ru: NDArray, rv: NDArray, hs: NDArray, const: CoreConst
+) -> NDArray:
+    """Calculate daily extraterrestrial solar radiation (J/m^2).
+
+    This function calculates the daily extraterrestrial solar radition (J/m^2)
+    using Eq. 1.10.3, :cite:t:'Duffie & Beckman:2013'
+
+    Args:
+        dr          : dimensionless distance factor
+        ru          : dimensionless variable substitute
+        rv          : dimensionless variable substitute
+        hs          : local hour angle, degrees
+        const       : CoreConst object containing core constants, including:
+            k_pir   : radians to degrees conversion, degrees/rad
+            k_secs_d: seconds in one solar day
+            k_Gsc   : planetary radiation constant, W/m^2
```

Comment:
Odd indenting on these rows.

I guess as a style, I've not tended to use spaces to align the argument descriptions.

Reply:
sorted